kernel/src/hil/text_screen.rs

```diff
@@ -0,0 +1,99 @@
+
+    /// Sends to the driver a command to hide the cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn hide_cursor(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to show the cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn show_cursor(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn on the blinking cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn blink_cursor_on(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn off the blinking cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn blink_cursor_off(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn on the display of the screen.
+    /// When finised, the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn display_on(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn off the display of the screen.
+    /// When finised, the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn display_off(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to cleat the display of the screen.
+    /// When finised, the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn clear(&self) -> ReturnCode;
+}
+
+pub trait TextScreenClient {
+    /// The driver calls this function when any command (but a write one)
+    /// finishes executing.
+    fn command_complete(&self, r: ReturnCode);
+
+    /// The driver calls this function when a write command finishes executing.
+    fn write_complete(&self, buffer: &'static mut [u8], r: ReturnCode);
+
+    /// The driver calls this function when the screen is ready to use after power on.
+    fn text_screen_is_ready(&self);
```

Comment:
This callback is a bit unexpected for two reasons. One, there is no `init()` (or similar) function in the HIL. So there would have to be some external code path the causes this callback to occur, _and_ it would have to occur after `set_client()` is called. Two, it's not clear why a client would care about this event. Presumably the only thing stopping the display from being ready is something actually trying to use it, and if so, then the display can just configure itself the first time the client tries to use it. In the case some external event has to happen (like maybe a person has to push a button or something) then I could see this callback being useful.

I think for simplicity this callback can just be removed.

Reply:
Hi! So, I wrote this callback because of two reasons. First, I saw that the `screen` hil has a similar callback, also with no `init()` function. Also, these text screen usually have a power-on sequence that needs to be done before the user can interact with the screen. 

At the moment, there is a `init()` function in the capsule that implements this hil, function that is called from the `main.rs` file, after the `set_client()` is called and executed. 

When I wrote this, I was thinking it would be a good idea if there were a mechanism to stop the capsule from executing commands from the user until that power-on sequence was done, so the hil won't send any command to the screen until the `text_screen_is_ready()` is called. But I believe I could force the init sequence to be executed with the first command from the user.

Tell me what you think and thank you!

Reply:
Done, the screen now gets initialized when the first `display_on` command is issued.